codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -1165,4 +1132,36 @@ private boolean allowModifications() {
             return pendingIterations == 0;
         }
     }
+
+    /**
+     * Implementation of {@link PropertyKey} that specifies the index position of the property.
+     */
+    class DefaultPropertyKey implements PropertyKey {
+        private final int index;
+
+        DefaultPropertyKey(int index) {
+            this.index = index;
+        }
+    }
+
+    /**
+     * A registry of all stream property keys known by this connection.
+     */
+    private class PropertyKeyRegistry {
+        final List<DefaultPropertyKey> keys = new ArrayList<DefaultPropertyKey>(4);
```

Comment:
@buchgr - I agree, but the concern is not necessarily with the memory keys use it is more with the number of extra allocations and the API's requirement that anything using keys also must not be static.  Just had a discussion with @nmittler.  There are a few other cleanup related items but for this PR we are not going to worry about this....in summary this is an improvement and we can optimize later if necessary.